master/buildbot/www/service.py

```diff
@@ -132,6 +133,12 @@ def uid(self):
         return jwt.encode(claims, self.site.session_secret, algorithm=SESSION_SECRET_ALGORITHM)
 
 
+class BuildbotLogFile(LogFile):
```

Comment:
This is definitely needed.  That's why I added it.
Without it, the Twisted logger opens in text-mode `"w"`, but there are many places
where the code tries to write bytes to the logger, and things break.
This behavior in the Twisted logger cannot be easily changed or patched.

That's why I went with the subclass of the Logger class.